src/test/java/teammates/test/cases/webapi/GetFeedbackResponsesActionTest.java

```diff
@@ -23,13 +45,197 @@ protected String getRequestMethod() {
+        loginAsStudent(student1InCourse1.getGoogleId());
+
+        ______TS("Not enough parameters");
+        verifyHttpParameterFailure();
+        verifyHttpParameterFailure(Const.ParamsNames.FEEDBACK_QUESTION_ID, qn1InSession1InCourse1.getId());
+        verifyHttpParameterFailure(Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString());
+    }
+
+    @Test
+    protected void testExecute_invalidIntent_shouldFail() {
+        loginAsStudent(student1InCourse1.getGoogleId());
+
+        ______TS("Invalid Intent");
+        String[] paramsForInvalidIntent = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn1InSession1InCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.FULL_DETAIL.toString(),
+        };
+        verifyHttpParameterFailure(paramsForInvalidIntent);
+    }
+
+    @Test
+    protected void testExecute_studentSubmission_shouldGetResponseSuccessfully() {
+        loginAsStudent(student1InCourse1.getGoogleId());
+
+        ______TS("Typical success case as a student");
+        String[] params = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn1InSession1InCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+        };
+        FeedbackResponsesData actualData = getFeedbackResponse(params);
+        List<FeedbackResponseData> actualResponses = actualData.getResponses();
+        assertEquals(1, actualResponses.size());
+        FeedbackResponseData actualResponse = actualResponses.get(0);
+        FeedbackResponseAttributes expected =
+                logic.getFeedbackResponsesFromStudentOrTeamForQuestion(qn1InSession1InCourse1,
+                        student1InCourse1).get(0);
+        assertNotNull(actualResponse.getFeedbackResponseId());
+        verifyFeedbackResponseEquals(expected, actualResponse);
+    }
+
+    @Test
+    protected void testExecute_instructorSubmission_shouldGetResponseSuccessfully() {
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+        ______TS("Typical success case as a instructor");
+        String[] params = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn2InGracePeriodInCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+        };
+        FeedbackResponsesData actualData = getFeedbackResponse(params);
+        List<FeedbackResponseData> actualResponses = actualData.getResponses();
+        assertEquals(1, actualResponses.size());
+
+        FeedbackResponseData actualResponse = actualResponses.get(0);
+        FeedbackResponseAttributes expected =
+                logic.getFeedbackResponsesFromInstructorForQuestion(qn2InGracePeriodInCourse1, instructor1OfCourse1)
+                        .get(0);
+        assertNotNull(actualResponse.getFeedbackResponseId());
+        verifyFeedbackResponseEquals(expected, actualResponse);
     }
 
     @Test
     @Override
     protected void testAccessControl() throws Exception {
```

Comment:
Similar to the `execute`, it's better to split test cases into different methods here as well.